New mHTGR griffin model (#158)
idaholab/virtual_test_bed: doc/content/htgr/mhtgr_griffin/mhtgr350_model_inputs.md

# Model Inputs

> The Griffin neutronics input deck for Excercise 1 is described below.

Reviewer: ```suggestion
The Griffin neutronics input file for Exercise 1 is described below.
```
no such thing as a deck anymore

user-desired information.
The cross sections for this model are not parameterized, instead, there only
exists one state point -- provided for the benchmark.
The power for the MHTGR is 350 MW and set with `tpow`.

### Global Parameters

Global parameters are parameters that may be referenced
throughout the input file.
This block is user specific; the purpose is
to simplify repeated variable entries.
However, be careful to not override a default parameter
option in a later block without meaning to.
It is common to provide cross section data for materials in
this block.
Here we include the number of energy groups for the diffusion solver with
[!style color=red](G) = 26.

!listing htgr/mhtgr_griffin/benchmark/griffin.i
         block=GlobalParams

## Geometry and Mesh

The mesh block reads in the provided Exodus mesh file.
Using the [!style color=orange](FileMeshGenerator) type, the mesh file
is identified with the [!style color=red](file) parameter.
It is common to read in the material IDs directly from the mesh if they are provided.
However, in this case, the mesh does not hold the material ids and these will be
specified in the `[Materials]` block.

!alert note

> Exodus files can be viewed with an Exodus supported visualization tool (i.e. ParaView).

Reviewer: ```suggestion
Exodus files can be viewed with an Exodus supported visualization tool (i.e. ParaView, standalone or through workbench, or peacock).
```

required to use the "PJFNKMO" method defined in the
`[Executioner]`.
This tells the finite element solver to not update the material
cross sections at each linear iteration.

!listing htgr/mhtgr_griffin/benchmark/griffin.i
         block=TransportSystems

## Executioner

The `[Executioner]` block lets the user specify the type of problem to solve.
Here, we select an [!style color=orange](Eigenvalue) problem
which will solve for the criticality and multi-group scalar flux
(eigenvalue and eigenvectors).
We also specify to solve with a Preconditioned Jacobian Free
Newton Krylov Matrix Only method by setting
[!style color=red](solve_type) equal to "PJFNKMO".
The Matrix Only method forces the solver to not update
material properties (i.e., cross sections) in the linear iterations
of the solve.
To use this solver, the [!style color=red](constant_matrices)
parameter must be set to "true".

There are optional arguments that may also be included.
For example, we first define a few PETSc options.
The maximum number of inner iterations is set with
[!style color=red](l_max_its) which is advised for a large number
of energy groups (26 in this case).

> The free power iteration scheme provides an educated initial guess of
> the solution.

Reviewer: ```suggestion
The initial free power iteration facilitate the elimination of higher order modes before
using a Newton-like method.
```

We set the number of [!style color=red](free_power_iterations) to 2.

!listing htgr/mhtgr_griffin/benchmark/griffin.i
         block=Executioner

### Post-processors, Debug, and Outputs

The last blocks are for post-processors, debug options, and outputs.
A post-processor can be thought of as a function to derive a variable
of interest from the solution.

> For example, the power density.

Reviewer: ```suggestion
For example, the total power is the integral of the power density over the entire spatial domain.
```

The blocks -- with fissile material -- at which the power density should
be computed are defined with [!style color=red](block).
Finally, the [!style color=red](execute_on) parameter
sets the time of execution.
In this case, the initial time step and at the end of each subsequent time step.

!listing htgr/mhtgr_griffin/benchmark/griffin.i
         block=Postprocessors

The debug options can be helpful when debugging a case.
These are a set of true (1) and false (0) options to print statements.

!listing htgr/mhtgr_griffin/benchmark/griffin.i
         block=Debug

The last block executes the output files from the simulation.
Two of the most common options include the exodus and csv file.
The Exodus file can be viewed with the same software as the mesh,
but now will show the solution and solution derived quantities such
as the multi-group scalar flux and power distribution.
The csv file stores a summary of the solution that includes
the criticality.
The [!style color=red](perf_graph) parameter is helpful to evaluate
the computational run time.

!listing htgr/mhtgr_griffin/benchmark/griffin.i
         block=Outputs

# How to run the model

> The model can be run with Griffin in serial or parallel.

Reviewer: ```suggestion
The model can be run with Griffin in serial or parallel respectively with:
```

> Run it via:

Reviewer: ```suggestion
```
